cpp/src/Slice/Parser.cpp

```diff
@@ -133,36 +209,43 @@ Slice::DefinitionContext::setSeenDefinition()
     _seenDefinition = true;
 }
 
-bool
-Slice::DefinitionContext::hasMetadata(const string& directive) const
+MetadataList
+Slice::DefinitionContext::getMetadata() const
 {
-    return findMetadata(directive).has_value();
+    return _metadata;
 }
 
 void
-Slice::DefinitionContext::setMetadata(const StringList& metadata)
+Slice::DefinitionContext::setMetadata(MetadataList metadata)
```

Comment:
Missing std::move below. Check the other setMetadata too.

Reply:
I checked through all the usage of `setMetadata` and added `move`s to the call-sites to avoid copies.